src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3109,6 +3158,7 @@ internal Buffer(IEnumerable<TElement> source)
                     count++;
                 }
             }
+
             this.items = items;
```

Comment:
There is a "new TElement[0];"   in the ToArray() implementation. Are we not using Array.Empty there for compat reasons?